tests/src/main/java/com/hazelcast/simulator/tests/topic/ReliableTopicTest.java

```diff
@@ -0,0 +1,168 @@
+package com.hazelcast.simulator.tests.topic;
+
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IAtomicLong;
+import com.hazelcast.core.ITopic;
+import com.hazelcast.core.Message;
+import com.hazelcast.core.MessageListener;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.simulator.test.TestContext;
+import com.hazelcast.simulator.test.annotations.Run;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Verify;
+import com.hazelcast.simulator.utils.AssertTask;
+import com.hazelcast.simulator.utils.ThreadSpawner;
+import com.hazelcast.topic.ReliableMessageListener;
+
+import static com.hazelcast.simulator.utils.CommonUtils.sleepRandomNanos;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+
+import static com.hazelcast.simulator.utils.TestUtils.assertTrueEventually;
+import static org.junit.Assert.assertEquals;
+
+public class ReliableTopicTest {
+
+    private static final ILogger LOGGER = Logger.getLogger(ReliableTopicTest.class);
+
+    // properties
+    public int topicCount = 10;
+    public int threadCount = 3;
+    public int listenersPerTopic = 2;
+    public String basename = "reliableTopic";
+
+
+    private IAtomicLong totalExpectedCounter;
+    private ITopic[] topics;
+    private TestContext testContext;
+    private HazelcastInstance hz;
+    private List<StressMessageListener> listeners;
+
+    @Setup
+    public void setup(TestContext testContext) throws Exception {
+        this.testContext = testContext;
+        hz = testContext.getTargetInstance();
+
+        totalExpectedCounter = hz.getAtomicLong(testContext.getTestId() + ":TotalExpectedCounter");
+        topics = new ITopic[topicCount];
+        listeners = new LinkedList<StressMessageListener>();
+        int listenerIdCounter = 0;
+        for (int k = 0; k < topics.length; k++) {
+            ITopic<Long> topic = hz.getReliableTopic(basename + "-" + k);
+            topics[k] = topic;
+            for (int l = 0; l < listenersPerTopic; l++) {
+                StressMessageListener topicListener = new StressMessageListener(listenerIdCounter);
+                listenerIdCounter++;
+                topic.addMessageListener(topicListener);
+                listeners.add(topicListener);
+            }
+        }
+    }
+
+    @Run
+    public void run() {
+        ThreadSpawner spawner = new ThreadSpawner(testContext.getTestId());
+        for (int k = 0; k < threadCount; k++) {
+            spawner.spawn(new Worker());
+        }
+        spawner.awaitCompletion();
+    }
+
+    private class Worker implements Runnable {
```

Comment:
Would it not be an idea to create a message which contains the thread and the sequence for that topic by that threads. Than you can do a more in depth analysis at the listener. Because it knows that it will get requests from a thread always in a specific order.